compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -163,16 +166,67 @@ generateSrcFromLf env = noLoc mod
             lname = mkRdrName (LF.unExprValName lfName) :: Located RdrName
             sig = TypeSig noExt [lname] . HsWC noExt . HsIB noExt <$> ltype
             lsigD = noLoc . SigD noExt <$> sig :: Gen (LHsDecl GhcPs)
-            lexpr = noLoc $ HsPar noExt $ noLoc $ HsVar noExt lname :: LHsExpr GhcPs
-            lgrhs = noLoc $ GRHS noExt [] lexpr :: LGRHS GhcPs (LHsExpr GhcPs)
-            grhss = GRHSs noExt [lgrhs] (noLoc $ EmptyLocalBinds noExt)
-            matchContext = FunRhs lname Prefix NoSrcStrict
-            lmatch = noLoc $ Match noExt matchContext [] Nothing grhss
-            lalts = noLoc [lmatch]
-            bind = FunBind noExt lname (MG noExt lalts Generated) WpHole []
+            bind = mkTrivialBind lname
             lvalD = noLoc $ ValD noExt bind :: LHsDecl GhcPs
         [ lsigD, pure lvalD ]
 
+    -- | Generate instance declarations from dictionary functions.
+    instanceDecls :: [Gen (LHsDecl GhcPs)]
+    instanceDecls = do
+        LF.DefValue {..} <- NM.toList $ LF.moduleValues $ envMod env
+        let dvalType = snd dvalBinder
+        Just qname@LF.Qualified{..} <- [getDFunClassName dvalType]
+        Just fieldNames <- [getDFunFieldNames dvalBody]
+        guard (not (isHasField qname))
+        pure $ do
+            polyTy <- HsIB noExt . noLoc <$> convType env dvalType
+            ghcMod <- genModule env qualPackage qualModule
+            pure . noLoc . InstD noExt . ClsInstD noExt $ ClsInstDecl
+                { cid_ext = noExt
+                , cid_poly_ty = polyTy
+                , cid_binds = listToBag (mapMaybe (mkBind ghcMod) fieldNames)
```

Comment:
Is there a reason why we don’t just generate an empty instance?

Reply:
Ah, I just assumed that wouldn't work. I'll try it out.

Reply:
We probably want to silence the warning but apart from that I don’t see why it shouldn’t work.